src/network/peloton_server.cpp

```diff
@@ -255,20 +254,26 @@ PelotonServer &PelotonServer::SetupServer() {
   int reuse = 1;
   setsockopt(listen_fd_, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(reuse));
 
-  TrySslOperation<int, const struct sockaddr *, socklen_t>(bind,
-                                                           listen_fd_,
-                                                           (struct sockaddr *) &sin,
-                                                           sizeof(sin));
+  TrySslOperation<int, const struct sockaddr *, socklen_t>(
+      bind, listen_fd_, (struct sockaddr *) &sin, sizeof(sin));
   TrySslOperation<int, int>(listen, listen_fd_, conn_backlog);
 
   dispatcher_task_ = std::make_shared<ConnectionDispatcherTask>(
       CONNECTION_THREAD_COUNT, listen_fd_);
 
-  LOG_INFO("Listening on port %llu", (unsigned long long)port_);
+  LOG_INFO("Listening on port %llu", (unsigned long long) port_);
   return *this;
 }
 
 void PelotonServer::ServerLoop() {
+  if (settings::SettingsManager::GetBool(settings::SettingId::rpc_enabled)) {
+    int rpc_port =
+        settings::SettingsManager::GetInt(settings::SettingId::rpc_port);
+    auto rpc_task = std::make_shared<PelotonRpcHandlerTask>(("127.0.0.1:"
+        + std::to_string(rpc_port)).c_str());
+    DedicatedThreadRegistry::GetInstance()
+        .RegisterDedicatedThread<PelotonRpcHandlerTask>(this, rpc_task);
```

Comment:
Is this RPC thread same to the listener thread?

Reply:
What listener?

Reply:
peloton port listener

Reply:
It listens on a different port. Are you talking about a specific class or anything?